pkg/resources/schema.go

```diff
@@ -208,94 +253,139 @@ func CreateContextSchema(ctx context.Context, d *schema.ResourceData, meta any)
+			return diag.FromErr(err)
 		}
-		return diag.FromErr(err)
-	}
-	if err := d.Set("name", schema.Name); err != nil {
-		return diag.FromErr(err)
-	}
 
-	if err := d.Set("is_transient", schema.IsTransient()); err != nil {
-		return diag.FromErr(err)
-	}
+		if err := d.Set("database", schema.DatabaseName); err != nil {
+			return diag.FromErr(err)
+		}
 
-	if err := d.Set("with_managed_access", schema.IsManagedAccess()); err != nil {
-		return diag.FromErr(err)
-	}
+		if err := d.Set("comment", schema.Comment); err != nil {
+			return diag.FromErr(err)
+		}
 
-	if err := d.Set("database", schema.DatabaseName); err != nil {
-		return diag.FromErr(err)
-	}
+		schemaParameters, err := client.Parameters.ShowParameters(ctx, &sdk.ShowParametersOptions{
+			In: &sdk.ParametersIn{
+				Schema: id,
+			},
+		})
+		if err != nil {
+			return diag.FromErr(err)
+		}
 
-	if err := d.Set("comment", schema.Comment); err != nil {
-		return diag.FromErr(err)
-	}
+		if diags := HandleDatabaseParameterRead(d, schemaParameters); diags != nil {
+			return diags
+		}
+		pipeExecutionPaused, err := collections.FindOne(schemaParameters, func(property *sdk.Parameter) bool {
+			return property.Key == "PIPE_EXECUTION_PAUSED"
+		})
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("failed to find schema PIPE_EXECUTION_PAUSED parameter, err = %w", err))
+		}
+		value, err := strconv.ParseBool((*pipeExecutionPaused).Value)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set(strings.ToLower(string(sdk.ObjectParameterPipeExecutionPaused)), value); err != nil {
+			return diag.FromErr(err)
+		}
 
-	schemaParameters, err := client.Parameters.ShowParameters(ctx, &sdk.ShowParametersOptions{
-		In: &sdk.ParametersIn{
-			Schema: id,
-		},
-	})
-	if err != nil {
-		return diag.FromErr(err)
-	}
+		describeResult, err := client.Schemas.Describe(ctx, schema.ID())
+		if err != nil {
+			return diag.FromErr(err)
+		}
 
-	if diags := HandleDatabaseParameterRead(d, schemaParameters); diags != nil {
-		return diags
-	}
-	pipeExecutionPaused, err := collections.FindOne(schemaParameters, func(property *sdk.Parameter) bool {
-		return property.Key == "PIPE_EXECUTION_PAUSED"
-	})
-	if err != nil {
-		return diag.FromErr(fmt.Errorf("failed to find schema PIPE_EXECUTION_PAUSED parameter, err = %w", err))
-	}
-	value, err := strconv.ParseBool((*pipeExecutionPaused).Value)
-	if err != nil {
-		return diag.FromErr(err)
-	}
-	if err := d.Set("pipe_execution_paused", value); err != nil {
-		return diag.FromErr(err)
-	}
+		if withExternalChangesMarking {
+			if err = handleExternalChangesToObjectInShow(d,
+				showMapping{"options", "is_transient", schema.IsTransient(), fmt.Sprintf("%t", schema.IsTransient()), func(x any) any {
+					return slices.Contains(sdk.ParseCommaSeparatedStringArray(x.(string), false), "TRANSIENT")
+				}},
+				showMapping{"options", "with_managed_access", schema.IsManagedAccess(), fmt.Sprintf("%t", schema.IsManagedAccess()), func(x any) any {
```

Comment:
same here, predefined mapping functions can be used

Reply:
Done.